### DISCUSSION TITLE
Many step sensor problems (ThrottlingException)

### DISCUSSION THREAD

**Discussion by dmlansans01** [apache/airflow#28916] [Q&A]
After adding many steps, I added the same sensor as the step to track the step. but at irregular intervals
"botocore.exceptions.ClientError: An error occurred (ThrottlingException) when calling the DescribeStep operation (reached max retries: 4): Rate exceeded"
An error occurs.
This seems to have occurred by using more than 10 per second in the "The maximum number of DescribeStep API requests that you can make per second" item in the "Amazone EMR" Service in "aws quotas"...
I wonder if StepSensorOperator has a function to retry instead of going to the next step in a failed state when an Exception occurs instead of a Fail. Airflow 1.10.14 keeps failing

**Comment by Taragolis** [comment]
[Amazon Web Services Connection: Avoid Throttling exceptions](https://airflow.apache.org/docs/apache-airflow-providers-amazon/stable/connections/aws.html#avoid-throttling-exceptions)

I do not know is `config_kwargs` could be set in Airflow 1.10.x ([EOL almost 2 year](https://endoflife.date/apache-airflow)) but you could try to set generic AWS Environment variables to achieve same stuff
